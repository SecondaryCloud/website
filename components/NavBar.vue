<template>
  <header class="navbar">
    <NuxtLink class="navbar__logo" to="/">
      <img src="/logo.svg" alt="Secondary - Page d'accueil" />
    </NuxtLink>
    <nav class="navbar__nav">
      <ul>
        <li>
          <NuxtLink to="/">Accueil</NuxtLink>
        </li>
        <li>
          <NuxtLink to="/tarif">Tarification</NuxtLink>
        </li>
      </ul>
    </nav>
  </header>
</template>

<script>
export default {
  name: 'NavBar',
}
</script>

<style lang="scss" scoped>
.navbar {
  display: flex;
  height: 60px;
  align-items: center;

  &__logo {
    height: 57px;
    width: 170px;
  }

  &__nav {
    font-family: $font-roboto;
    font-size: 1rem;
    flex: 1;
    display: flex;
    justify-content: center;

    ul {
      list-style: none;
      display: flex;
      padding: 0;
    }

    a {
      text-decoration: none;
      color: $eden;
      padding: 0 8px;

      &:visited {
        color: $eden;
      }
    }
  }
}
</style>
