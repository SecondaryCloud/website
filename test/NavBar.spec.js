import { mount, shallowMount } from '@vue/test-utils'
import NavBar from '@/components/NavBar.vue'

describe('NavBar', () => {
  test('is a Vue instance', () => {
    const wrapper = shallowMount(NavBar, {
      stubs: {
        NuxtLink: true,
      },
    })
    expect(wrapper.vm).toBeTruthy()
  })

  test('render properly', () => {
    const component = mount(NavBar)
    expect(component.html()).toMatchSnapshot()
  })
})
