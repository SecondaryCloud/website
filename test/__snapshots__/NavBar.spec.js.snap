// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar render properly 1`] = `
"<header class=\\"navbar\\">
  <nuxtlink to=\\"/\\" class=\\"navbar__logo\\"><img src=\\"/logo.svg\\" alt=\\"Secondary - Page d'accueil\\"></nuxtlink>
  <nav class=\\"navbar__nav\\">
    <ul>
      <li>
        <nuxtlink to=\\"/\\">Accueil</nuxtlink>
      </li>
      <li>
        <nuxtlink to=\\"/tarif\\">Tarification</nuxtlink>
      </li>
    </ul>
  </nav>
</header>"
`;
